Custom task problemMatcher results not appearing in problems panel when problem is on the first line

This may be an insufficient documentation bug. I've followed the documentation here:
https://code.visualstudio.com/docs/editor/tasks#_defining-a-problem-matcher

But I am unable to get problems to appear. I've created a minimal example to reproduce this.
See here: https://github.com/xxgreg/vscode-problem-matcher-example

You can either check this out, or follow the reproduction instructions below.

Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 
```
code --version
1.62.0
b3318bc0524af3d74034b8bb8a64df0ccf35549a
x64
```

- OS Version:
MacOS 11.6

Steps to Reproduce:

Create a new workspace, and open it.

Add `.vscode/tasks.json`
```
{
    "version": "2.0.0",
    "tasks": [
        {
            "label": "make an error",
            "command": "echo",
            "args": ["blah.js:4:3: warning: your foo is bad"],
            "problemMatcher": {
              "fileLocation": ["relative", "${workspaceFolder}"],
              "pattern": {
                "regexp": "^(.*):(\\d+):(\\d+):\\s+(warning|error):\\s+(.*)$",
                "file": 1,
                "line": 2,
                "column": 3,
                "severity": 4,
                "message": 5
              }
            }
          }
    ]
}
```

Add a file `blah.js`
```


// Output an error for this line:
var bad = "foo";

```

Run the task "make an error".

Observe that no problem appears in the problems view, or inline in `blah.js`.

So it looks like the newline handling doesn't work correctly for errors on the first line of the output.

If I replace echo with the following go program it works correctly.

```
package main

import (
	"fmt"
	"os"
)

func main() {
	fmt.Fprintln(os.Stderr, "\n\nblah.js:5:3: warning: implicit declaration of function ‘prinft’\n\n")
}
```

--- a/src/vs/workbench/contrib/tasks/browser/terminalTaskSystem.ts
+++ b/src/vs/workbench/contrib/tasks/browser/terminalTaskSystem.ts
@@ -828,12 +828,7 @@ export class TerminalTaskSystem extends Disposable implements ITaskSystem {
 				this.logService.error('Task terminal process never got ready');
 			});
 			this.fireTaskEvent(TaskEvent.create(TaskEventKind.Start, task, terminal.instanceId));
-			let skipLine: boolean = (!!task.command.presentation && task.command.presentation.echo);
 			const onData = terminal.onLineData((line) => {
-				if (skipLine) {
-					skipLine = false;
-					return;
-				}
 				watchingProblemMatcher.processLine(line);
 				if (!delayer) {
 					delayer = new Async.Delayer(3000);
@@ -919,12 +914,7 @@ export class TerminalTaskSystem extends Disposable implements ITaskSystem {
 			let problemMatchers = await this.resolveMatchers(resolver, task.configurationProperties.problemMatchers);
 			let startStopProblemMatcher = new StartStopProblemCollector(problemMatchers, this.markerService, this.modelService, ProblemHandlingStrategy.Clean, this.fileService);
 			this.terminalStatusManager.addTerminal(task, terminal, startStopProblemMatcher);
-			let skipLine: boolean = (!!task.command.presentation && task.command.presentation.echo);
 			const onData = terminal.onLineData((line) => {
-				if (skipLine) {
-					skipLine = false;
-					return;
-				}
 				startStopProblemMatcher.processLine(line);
 			});
 			promise = new Promise<ITaskSummary>((resolve, reject) => {

